sktime/transformations/series/peak.py

```diff
@@ -0,0 +1,441 @@
+        keep_original_columns=False,
+        keep_original_PeakTimeFeature_columns=False,
+    ):
+        self.ts_freq = ts_freq
+        self.peak_hour_start = peak_hour_start
+        self.peak_hour_end = peak_hour_end
+        self.peak_day_start = peak_day_start
+        self.peak_day_end = peak_day_end
+        self.peak_week_start = peak_week_start
+        self.peak_week_end = peak_week_end
+        self.peak_month_start = peak_month_start
+        self.peak_month_end = peak_month_end
+        self.peak_quarter_start = peak_quarter_start
+        self.peak_quarter_end = peak_quarter_end
+        self.peak_year_start = peak_year_start
+        self.peak_year_end = peak_year_end
+        self.working_hour_start = working_hour_start
+        self.working_hour_end = working_hour_end
+        self.keep_original_columns = keep_original_columns
+        self.keep_original_PeakTimeFeature_columns = (
+            keep_original_PeakTimeFeature_columns
+        )
+
+        super().__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing the core logic, called from transform
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+        Returns
+        -------
+        Xt : pd.Series or pd.DataFrame, same type as X
+            transformed version of X
+        """
+        _check_inputs(self.peak_hour_start, self.peak_hour_end, "peak_hour", 0, 23)
+        _check_inputs(self.peak_day_start, self.peak_day_end, "peak_day", 0, 6)
+        _check_inputs(self.peak_week_start, self.peak_week_end, "peak_week", 1, 53)
+        _check_inputs(self.peak_month_start, self.peak_month_end, "peak_month", 1, 12)
+        _check_inputs(
+            self.peak_quarter_start, self.peak_quarter_end, "peak_quarter", 1, 4
+        )
+        _check_inputs(self.peak_year_start, self.peak_year_end, "peak_year", 1900, 2100)
+        _check_inputs(
+            self.working_hour_start, self.working_hour_end, "working_hour", 0, 23
+        )
+
+        if isinstance(X.index, pd.MultiIndex):
+            time_index = X.index.get_level_values(-1)
+        else:
+            time_index = X.index
+
+        x_df = pd.DataFrame(index=X.index)
+        if isinstance(time_index, pd.PeriodIndex):
+            x_df["date_sequence"] = time_index.to_timestamp()
+        elif isinstance(time_index, pd.DatetimeIndex):
+            x_df["date_sequence"] = time_index
+        else:
+            raise ValueError("Index type not supported")
+
+        cd = _extract_datetime_features(x_df)
+
+        datetime_freq = _datetime_frequency_rank()
+
+        # check time series freq
+        _check_ts_freq(x_df, datetime_freq, self.ts_freq)
+
+        df = _extract_peaktime_features(self, cd, datetime_freq)
```

Comment:
```suggestion
        df = self._extract_peaktime_features(cd, datetime_freq)
```

Why not something like this?

Reply:
Done!